src/App.tsx

```diff
@@ -1,10 +1,50 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getTodos } from './api/todos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Todolist } from './components/Todolist';
+import { filterValues } from './constants';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+const USER_ID = 6438;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorType, setErrorType] = useState('');
+  const [hasError, setHasError] = useState(false);
+  const [hasCompleted, setHasCompleted] = useState(false);
+  const [selectedFilter, setSelectedFilter] = useState('All');
+  const hasActive = todos.some(todoItem => !todoItem.completed);
+
+  const filteredTodos = todos.filter(todo => {
+    if (selectedFilter === filterValues.completed) {
```

Comment:
use enum in all cases
```suggestion
    if (selectedFilter === FilterValues.completed) {
```

Reply:
got it